code/modules/ninja/martial_art.dm

```diff
@@ -0,0 +1,181 @@
+			if(G == A.get_active_hand() && G.state >= GRAB_NECK) // Grab must be in the user's active hand for the duration of the strangle.
+				D.silent += 1
+				D.adjustOxyLoss(1)
+			else
+				D.visible_message("<span class='warning'>[A] loses his grip on [D]'s neck!</span>", \
+									"<span class='userdanger'>[A] loses his grip on your neck!</span>")
+				has_choke_hold = 0
+				return 0
+			I++
+			sleep(5)
+
+		A << "<span class='warning'>You feel [D] go limp in your grip.</span>"
+		D << "<span class='userdanger'>You feel your consciousness slip away as [A] strangles you!</span>"
+		D.sleeping += 20
+
+		has_choke_hold = 0
+
+		return 1
+	return A.pointed(D)
+
+/datum/martial_art/ninja_martial_art/proc/palm_strike(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	if(!D.stat && !D.weakened)
+		if(has_focus)
+			has_focus = 0
+			A.face_atom(D)
+
+			var/strike_adjective = pick(attack_names)
+			D.visible_message("<span class='danger'>[A] sends [D] flying backwards with a [strike_adjective] palm strike!</span>", \
+				"<span class='userdanger'>[A] delivers a [strike_adjective] palm strike to you and sends you flying!</span>")
+
+			var/atom/throw_target = get_ranged_target_turf(D, get_dir(D, get_step_away(D, A)), 3) // Get a turf 3 tiles away from the target relative to our direction from him.
+			D.throw_at(throw_target, 200, 4) // Throw the poor bastard at the target we just gabbed.
+
+			D.Weaken(2)
+			playsound(get_turf(D), 'sound/weapons/punch1.ogg', 50, 1, -1)
+			spawn(50) has_focus = 1
+			return 1
+		A << "<span class='warning'>You are not focused enough to use that move yet!</span>"
+		return 0
+	return A.pointed(D)
+
+/datum/martial_art/ninja_martial_art/grab_act(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D) //Instant aggressive grab
+	D.grabbedby(A)
+	var/obj/item/weapon/grab/G = A.get_active_hand()
+	if(G)
+		G.state = GRAB_AGGRESSIVE
+
+	return 1
+
+/datum/martial_art/ninja_martial_art/harm_act(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D) // 10 damage punches
+	var/strike_name = "[pick(attack_names)] [pick("punches", "kicks", "chops", "slams", "strikes")]"
+	D.visible_message("<span class='danger'>[A] [strike_name] on [D]!</span>", \
+					  "<span class='userdanger'>[A] [strike_name] you!</span>")
+	D.apply_damage(10, BRUTE)
+	playsound(get_turf(D), 'sound/weapons/punch1.ogg', 50, 1, -1)
+	return 1
+
+// Ninja middle click override, required for the special moves to function and handled in grant_ninja_martial_art
+
+/datum/middleClickOverride/ninja_martial_art
+
+/datum/middleClickOverride/ninja_martial_art/onClick(var/atom/A, var/mob/living/carbon/human/user)
+	if(!user.martial_art || user.martial_art.name != "Creeping Widow Style") // Checking based on name because I'm fairly certain that checking against a user's specific ninja martial arts datum versus a generic one won't work.
```

Comment:
You could just check, `if(!istype(user.martial_art, /datum/martial_art/ninja_martial_art)`